tests/test_router_handover_leak.cpp

```diff
@@ -0,0 +1,164 @@
+/*
```

Comment:
I am not sure if this test makes sense. It demonstrates the leak, but it doesn't assert anything related to the sockets, so it probably won't fail if the leak is present (or it will fail because resources are exhausted). It probably consumes quite a lot of time compared to other tests, provided it does so many iterations. 

Unfortunately, leaking file descriptors cannot be detected in the same way as leaking memory with valgrind. valgrind can show open file descriptors at exit with the --track-fds option, but it will only report them. The output of valgrind would need to be checked for that explicitly.

If this test case is really added, please use the Unity test framework, and the assertion macros like in test_router_handover. You can add it to the existing test_router_handover.cpp file.

Reply:
> so it probably won't fail if the leak is present (or it will fail because resources are exhausted).

Exactly, it will fail because resources are exhausted. And why this is bad?

> It probably consumes quite a lot of time compared to other tests, provided it does so many iterations.

Good pass:
```
$ time ./tests/test_router_handover_leak 
Total fds: 18
Total fds: 18
Total fds: 18
Total fds: 18
OK

real    0m0.948s
user    0m0.496s
sys     0m0.719s
```

Bad pass:
```
$ time ./tests/test_router_handover_leak 
Total fds: 1018
lt-test_router_handover_leak: tests/test_router_handover_leak.cpp:103: int main(int, char**): Assertion `dealer_two' failed.
Aborted (core dumped)

real    0m0.767s
user    0m0.133s
sys     0m0.340s
```

Less than 1 second for both runs.

And that's how bad pass looks when I run 'make check':

```
./config/test-driver: line 107:  5975 Aborted                 (core dumped) "$@" > $log_file 2>&1
FAIL: tests/test_router_handover_leak
```

>Unfortunately, leaking file descriptors cannot be detected in the same way as leaking memory with valgrind.

I do not need valgrind for that, why? The test simply crashes if we reach soft kernel limit of fds (I am not sure what happens on windows, but there should be something similar, right? Otherwise you can exhaust kernel resources from user app).

The other simple way is to check number of descriptors and expect they won't grow up to, say, 50. I did not want to do that, because I do not know how to do that on windows, so I wanted to keep code less os specific, thus test just crashes.

> please use the Unity test framework, and the assertion macros like in test_router_handover.

sure

> You can add it to the existing test_router_handover.cpp file.

I do not want to mix leak testing with generic handover test.

Reply:
> Exactly, it will fail because resources are exhausted. And why this is bad?

It is maybe not bad but less than optimal, because it is dependent on the environment (OS defaults and custom user limits) when it fails. I suspect that once the build is fine it will take much longer under Windows, but I am not sure.

If there is no viable alternative, it can be done this way, and maybe disable it on Windows if necessary.

I don't know if there is a documented way to check the number of open sockets in Windows.